plugins/adrv9009_adv.c

```diff
@@ -1013,107 +1030,164 @@ static int adrv9009adv_handle_driver(const char *attrib, const char *value)
+	return (gpointer)g_strdup(src);
 }
 
-static bool adrv9009adv_identify(void)
+static void context_destroy(struct osc_plugin *plugin, const char *ini_fn)
 {
-	/* Use the OSC's IIO context just to detect the devices */
-	struct iio_context *osc_ctx = get_context_from_osc();
+	save_profile(plugin, ini_fn);
 
-	dev = iio_context_find_device(osc_ctx, PHY_DEVICE);
+	size_t n = 0;
+	for (; n < plugin->priv->sr_attribs_count; n++) {
+		g_free(plugin->priv->sr_attribs[n]);
+	}
+	g_free(plugin->priv->sr_attribs);
 
-	return !!dev && iio_device_get_debug_attrs_count(dev);
+	osc_plugin_context_free_resources(&plugin->priv->plugin_ctx);
+	
+	osc_destroy_context(plugin->priv->ctx);
+	
+	g_free(plugin->priv);
 }
 
-struct osc_plugin plugin = {
-	.name = THIS_DRIVER,
-	.identify = adrv9009adv_identify,
-	.init = adrv9009adv_init,
-	.handle_item = adrv9009adv_handle,
-	.handle_external_request = handle_external_request,
-	.update_active_page = update_active_page,
-	.save_profile = save_profile,
-	.load_profile = load_profile,
-	.destroy = context_destroy,
-};
+struct osc_plugin * create_plugin(struct osc_plugin_context *plugin_ctx)
+{
+	struct osc_plugin *plugin= g_new0(struct osc_plugin, 1);
+
+	if (!plugin_ctx ) {
+		fprintf(stderr, "Cannot create plugin because an invalid plugin context was provided\n");
+		return NULL;
+	}
+
+	plugin->priv = g_new0(struct plugin_private, 1);
+	plugin->priv->plugin_ctx.plugin_name = g_strdup(plugin_ctx->plugin_name);
+	plugin->priv->plugin_ctx.required_devices = g_list_copy_deep(plugin_ctx->required_devices, (GCopyFunc)copy_gchar_array, NULL);
+
+	plugin->name = plugin->priv->plugin_ctx.plugin_name;
+	plugin->dynamically_created = TRUE;
+	plugin->init = adrv9009adv_init;
+	plugin->handle_item = adrv9009adv_handle;
+	plugin->handle_external_request = handle_external_request;
+	plugin->update_active_page = update_active_page;
+	plugin->save_profile = save_profile;
+	plugin->load_profile = load_profile;
+	plugin->destroy = context_destroy;
+
+	return plugin;
+}
+
+/* Informs how many plugins can be instantiated and gives context for each possible plugin instance */
+GArray* get_data_for_possible_plugin_instances(void)
+{
+	GArray *data = g_array_new(FALSE, TRUE, sizeof(struct osc_plugin_context *));
+	struct iio_context *osc_ctx = get_context_from_osc();
+	GArray *devices = get_iio_devices_starting_with(osc_ctx, PHY_DEVICE);
+	guint i = 0;
+
+	for (; i < devices->len; i++) {
+		struct osc_plugin_context *context = g_new(struct osc_plugin_context, 1);
+		struct iio_device *dev = g_array_index(devices, struct iio_device*, i);
+
+		/* Construct the name of the plugin */
+		char *name;
+		if (devices->len > 1)
+			name = g_strdup_printf("%s-%i", THIS_DRIVER, i);
+		else
+			name = g_strdup(THIS_DRIVER);
+
+		context->required_devices = NULL;
+		context->required_devices = g_list_append(context->required_devices, g_strdup(iio_device_get_name(dev)));
+		context->plugin_name = g_strdup(name);
+		g_free(name);
```

Comment:
why not just assign `name` here and remove `g_free` ?
one less `g_strdup()`

Reply:
That's better, thanks!